db/migrate/20241030210727_split_hide_profile_and_presence.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class SplitHideProfileAndPresence < ActiveRecord::Migration[7.1]
+  def up
+    # Split default_hide_profile_and_presence if setting exists
+    result =
+      execute(
+        "SELECT * FROM site_settings WHERE name = 'default_hide_profile_and_presence' LIMIT 1",
```

Comment:
Almost nitpicky, but since we only care about those two values, it's safer to select just those two in the SQL.

```suggestion
        "SELECT value, data_type FROM site_settings WHERE name = 'default_hide_profile_and_presence' LIMIT 1",
```